contracts/wawaka/common/Value.cpp

```diff
@@ -114,47 +114,26 @@ bool ww::value::Value::serialize(StringArray& result) const
         return false;
     }
 
-    // serialize the result
-    size_t serialized_size = json_serialization_size(value_);
-    char *serialized_response = (char *)malloc(serialized_size + 1);
+    char *serialized_response = (char *)serialize();
     if (serialized_response == NULL)
-    {
-        CONTRACT_SAFE_LOG(1, "failed serialization; failed allocation");
         return false;
-    }
 
-    JSON_Status jret = json_serialize_to_buffer(value_, serialized_response, serialized_size + 1);
-    if (jret != JSONSuccess)
-    {
-        CONTRACT_SAFE_LOG(1, "failed serialization; invalid value");
-        free(serialized_response);
-        return false;
-    }
+    bool success = result.assign(serialized_response);
```

Comment:
I liked the `take` approach more. No copy needed.

Reply:
safety over performance in this case. easy enough to change later. but for the moment i'm working on the assumption that all free's should be very close to the corresponding malloc's.

Reply:
added a comment with reference to take.